gettsim/typing.py

```diff
@@ -0,0 +1,38 @@
+from typing import TypeVar
+
+import numpy as np
+
+FloatSeries = TypeVar("FloatSeries")
+IntSeries = TypeVar("IntSeries")
+BoolSeries = TypeVar("BoolSeries")
+DateTimeSeries = TypeVar("DateTimeSeries")
+
+
+def check_if_series_has_internal_type(series, internal_type):
+    """Check if data type of series fits to the internal type of gettsim.
+
+    Parameters
+    ----------
+    series : pd.Series
+        Some data series.
+    internal_type : TypeVar
+        One of the internal gettsim types.
+
+    Returns
+    -------
+    out : bool
+        Return check variable.
+    """
+    dtype = series.dtype
+    if internal_type == FloatSeries:
+        # I think we can allow for ints here?!
+        out = np.issubdtype(dtype, np.integer) or np.issubdtype(dtype, np.floating)
+    elif internal_type == BoolSeries:
+        out = np.issubdtype(dtype, np.bool)
+    elif internal_type == IntSeries:
```

Comment:
Make sure this covers both the old integer type and the new (experimental) pandas integer type (https://pandas.pydata.org/pandas-docs/stable/user_guide/integer_na.html). Not sure if it does right now.

Same question for the BoolSeries

Reply:
I changed to pandas checks throughout!